### DISCUSSION TITLE
HB and BOHB to support nested search queries

### DISCUSSION THREAD

**Discussion by H-arshit** [microsoft/nni#3816] [Ideas]

**Comment by QuanluZhang** [comment]
Thanks @H-arshit . We have not had plan for the support of nested search space for HB and BOHB, but we agree this is a good feature to have and will be appreciated if external contributors could contribute on this feature.

It is easy to support nested search space for HB, as trial configurations are randomly sampled from search space. Making the sampling aware of the hierarchy of the user specified search space.

@PurityFan Could you also give some suggestions about how to support nested search space on BOHB?

**Comment by PurityFan** [comment]
In BOHB, our search space  have to represent in ConfigSpace, currently this library doesn't seem to support nested values.
If we have to do this, we may hash the nested dict into a double value or choice. But this may be troublesome.
